altamisa/isatab/parse_investigation.py

```diff
@@ -389,7 +409,7 @@ def _read_ontology_source_reference(self) -> Iterator[
                 tpl = 'Incomplete ontology source; found: {}, {}, {}, {}'
                 msg = tpl.format(name, file_, version, desc)
                 raise ParseIsatabException(msg)
-            comments = {k: section[k][i] for k in comment_keys}
+            comments = self._parse_comments(section, comment_keys, i)
```

Comment:
Static methods should be called in a static way. Otherwise, consider making it a global function. Same below.